packages/exchange/src/pairs/Bridge.ts

```diff
@@ -0,0 +1,87 @@
+import Pair, { ExchangeParams, ValueTypes } from './Pair';
+import { Asset } from '@burner-wallet/types';
+import promiseRetry from 'promise-retry';
+import Web3 from "web3";
+const { toBN } = Web3.utils;
+
+
+interface BridgePairConstructor {
+    assetA: string;
+    assetABridge: string;
+    assetB: string;
+    assetBBridge: string;
+}
+
+interface BridgeExchangeParams {
+    asset: Asset;
+    assetOnOtherNetwork: Asset;
+    from: string;
+    value: string;
+    to: string;
+}
+
+export default class Bridge extends Pair {
+    protected readonly assetABridge: string;
+    protected readonly assetBBridge: string;
+
+    constructor({ assetA, assetABridge, assetB, assetBBridge }: BridgePairConstructor) {
+        super({ assetA, assetB });
+        this.assetABridge = assetABridge;
+        this.assetBBridge = assetBBridge;
+    }
+
+    exchangeAtoB({ account, value, ether }: ExchangeParams) {
+        const _value = this._getValue({ value, ether });
+        const asset = this.getExchange().getAsset(this.assetA);
+        const assetOnOtherNetwork = this.getExchange().getAsset(this.assetB);
+        return this._exchangeAndWait({
+            asset,
+            assetOnOtherNetwork,
+            from: account,
+            value: _value,
+            to: this.assetABridge
+        });
+    }
+
+    exchangeBtoA({ account, value, ether }: ExchangeParams) {
+        const _value = this._getValue({ value, ether });
+        const asset = this.getExchange().getAsset(this.assetB);
+        const assetOnOtherNetwork = this.getExchange().getAsset(this.assetA);
+        return this._exchangeAndWait({
+            asset,
+            assetOnOtherNetwork,
+            from: account,
+            value: _value,
+            to: this.assetBBridge
+        });
+    }
+
+    async _exchangeAndWait({ asset, assetOnOtherNetwork, from, value, to }: BridgeExchangeParams) {
+        const initialBalance = await assetOnOtherNetwork.getBalance(from);
+        const result = await asset.send({ from, value, to });
+        await promiseRetry(async (retry) => {
```

Comment:
I'd like to avoid introducing new dependencies if possible. Would this logic work the same using this?

```javascript
const wait = (time: number) => new Promise(resolve => setTimeout(resolve, time));

while (true) {
  if (/* logic */) {
    return result;
  }
  await wait(1200);
}
```

Reply:
Thanks! I applied your suggested logic in the methods and removed the `promise-retry` dependencies.